File: Coding Exercises/TypeScript/Day31/findYoungestBrother.ts
class Brother {
  constructor(public name: string, public age: number) {}
}

function findYoungestBrother(brothers: Brother[]): Brother | null {
  if (brothers.length === 0) {
    return null;
  }

  let youngest = brothers[0];

  for (const brother of brothers.slice(1)) {
    if (brother.age < youngest.age) {
      youngest = brother;
    }
  }

  return youngest;
}

File: Coding Exercises/Python/Day31/find_youngest_brother.py
class Brother:
    def __init__(self, name, age):
        self.name = name
        self.age = age


def find_youngest_brother(brothers):
    if not brothers:
        return None

    youngest = brothers[0]
    for brother in brothers[1:]:
        if brother.age < youngest.age:
            youngest = brother

    return youngest
